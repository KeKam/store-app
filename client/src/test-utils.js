import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';

import rootReducer from './redux/root-reducer';

// const render = (
//   component,
//   {
//     initialState,
//     store = createStore(rootReducer, initialState),
//     ...renderOptions
//   } = {}
// ) => {
//   return {
//     ...rtlRender(<Provider store={store}>{component}</Provider>),
//     store,
//   };
//   // const Wrapper = ({ children }) => {
//   //   return <Provider store={store}>{children}</Provider>;
//   // };
//   // return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
// };

const render = (component, { initialState } = {}) => {
  const actions = [];
  const observerMiddleware = () => (next) => (action) => {
    actions.push(action);
    return next(action);
  };
  const store = createStore(
    rootReducer,
    initialState,
    applyMiddleware(observerMiddleware)
  );
  const utils = {
    dispatch(action) {
      return store.dispatch(action);
    },
    getDispatchedActions() {
      return actions;
    },
    getState() {
      return store.getState();
    },
  };
  return {
    ...rtlRender(<Provider store={store}>{component}</Provider>),
    ...utils,
  };
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { render };
