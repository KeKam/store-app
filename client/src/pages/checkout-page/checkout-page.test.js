import React from 'react';
import '@testing-library/jest-dom';

import { render, screen } from '../../test-utils';
import CheckoutPage from './checkout-page';

describe('CheckoutPage component', () => {
  it('should render CheckoutPage component', () => {
    const { asFragment } = render(<CheckoutPage />);

    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText('Product')).toBeInTheDocument();
    expect(screen.getByText('Description')).toBeInTheDocument();
    expect(screen.getByText('Quantity')).toBeInTheDocument();
    expect(screen.getByText('Price')).toBeInTheDocument();
    expect(screen.getByText('Remove')).toBeInTheDocument();
    expect(screen.getByText('You have no added items')).toBeInTheDocument();
    expect(screen.getByTestId('cart-total')).toHaveTextContent('TOTAL : 0 €');
    expect(screen.getByTestId('warning')).toHaveTextContent(
      '*Please use this test credit card for payments on this site*4242 4242 4242 4242 - Exp: 01/21 - CVC: 543'
    );
    expect(screen.getByRole('button', { name: 'Pay Now' })).toBeInTheDocument();
  });

  it('should show CheckoutItem', () => {
    const mockState = {
      cartItems: [
        {
          id: 1,
          name: 'Winter falls',
          imageUrl: 'www.testImage.com',
          price: 100,
          quantity: 2,
        },
      ],
    };

    render(<CheckoutPage />, { initialState: { cart: mockState } });

    expect(screen.getByTestId('title')).toHaveTextContent('Winter falls');
    expect(screen.getByTestId('left-arrow')).toBeInTheDocument();
    expect(screen.getByTestId('quantity')).toHaveTextContent('2');
    expect(screen.getByTestId('right-arrow')).toBeInTheDocument();
    expect(screen.getByTestId('price')).toHaveTextContent('200 €');
    expect(screen.getByTestId('remove-button')).toBeInTheDocument();
    expect(screen.getByTestId('cart-total')).toHaveTextContent('TOTAL : 200 €');
    expect(
      screen.queryByText('You have no added items')
    ).not.toBeInTheDocument();
  });
});
