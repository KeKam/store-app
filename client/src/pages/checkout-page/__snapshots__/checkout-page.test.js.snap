// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutPage component should render CheckoutPage component 1`] = `
<DocumentFragment>
  <div
    class="sc-gzVnrw eIByyL"
  >
    <div
      class="sc-htoDjs kcQdPC"
    >
      <div
        class="sc-dnqmqq eUgIDW"
      >
        <span>
          Product
        </span>
      </div>
      <div
        class="sc-dnqmqq eUgIDW"
      >
        <span>
          Description
        </span>
      </div>
      <div
        class="sc-dnqmqq eUgIDW"
      >
        <span>
          Quantity
        </span>
      </div>
      <div
        class="sc-dnqmqq eUgIDW"
      >
        <span>
          Price
        </span>
      </div>
      <div
        class="sc-dnqmqq eUgIDW"
      >
        <span>
          Remove
        </span>
      </div>
    </div>
    <span
      class="sc-gqjmRU MHWQq"
    >
      You have no added items
    </span>
    <div
      class="sc-iwsKbI jpayIg"
    >
      <span
        data-testid="cart-total"
      >
        TOTAL : 0 â‚¬
      </span>
    </div>
    <div
      class="sc-gZMcBi hiZTFl"
      data-testid="warning"
    >
      *Please use this test credit card for payments on this site*
      <br />
      4242 4242 4242 4242 - Exp: 01/21 - CVC: 543
    </div>
    <button
      class="StripeCheckout"
      style="overflow: hidden; display: inline-block; border: 0px; padding: 1px; text-decoration: none; border-radius: 5px; box-shadow: 0 1px 0 rgba(0,0,0,0.2); cursor: pointer; visibility: visible; user-select: none;"
    >
      <span
        style="font-family: \\"Helvetica Neue\\",Helvetica,Arial,sans-serif; font-size: 14px; position: relative; padding: 0px 12px; display: block; height: 30px; line-height: 30px; color: rgb(255, 255, 255); font-weight: bold; box-shadow: inset 0 1px 0 rgba(255,255,255,0.25); text-shadow: 0 -1px 0 rgba(0,0,0,0.25); border-radius: 4px;"
      >
        Pay Now
      </span>
    </button>
  </div>
</DocumentFragment>
`;
